src/instructions/segmentation.rs

```diff
@@ -18,11 +18,14 @@ pub unsafe fn set_cs(sel: SegmentSelector) {
     #[cfg(feature = "inline_asm")]
     #[inline(always)]
     unsafe fn inner(sel: SegmentSelector) {
-        llvm_asm!("pushq $0; \
-              leaq  1f(%rip), %rax; \
-              pushq %rax; \
-              lretq; \
-              1:" :: "ri" (u64::from(sel.0)) : "rax" "memory");
+        // Returns
+        unsafe fn ret_sym() {
+            asm!("ret");
+        }
+
+        // Pushes the new CS and the address of the "ret" function
+        asm!("push {cs}; lea {out_reg}, [{ret_sym}]; push {out_reg}; retf",
+            cs = in(reg) u64::from(sel.0), ret_sym = sym ret_sym, out_reg = out(reg) _);
```

Comment:
I think this might be why we were seeing test failures.  We can just directly use the existing implementation with the new `asm!` syntax:
```rust
asm!(
    "push {cs}",
    "lea {tmp}, [1f + rip]",
    "push {tmp}",
    "retfq",
    "1:",
    cs = in(reg) u64::from(sel),
    tmp = lateout(reg) _,
    options(nomem, preserves_flags),
);
```

I compiled this locally, and got the same assembly output as the existing implementation.